src/include/type/value_factory.h

```diff
@@ -1,597 +1,599 @@
+        case TypeId::DECIMAL: {
+          if (value.GetAs<double>() > (double)PELOTON_INT64_MAX ||
+              value.GetAs<double>() < (double)PELOTON_INT64_MIN)
+            throw Exception(ExceptionType::OUT_OF_RANGE,
+                            "Numeric value out of range.");
+          return ValueFactory::GetBigIntValue((int64_t)value.GetAs<double>());
+        }
+        case TypeId::VARCHAR: {
+          std::string str = value.ToString();
+          int64_t bigint = 0;
+          try {
+            bigint = stoll(str);
+          } catch (std::out_of_range &e) {
+            throw Exception(ExceptionType::OUT_OF_RANGE,
+                            "Numeric value out of range.");
+          }
+          if (bigint > PELOTON_INT64_MAX || bigint < PELOTON_INT64_MIN)
+            throw Exception(ExceptionType::OUT_OF_RANGE,
+                            "Numeric value out of range.");
+          return ValueFactory::GetBigIntValue(bigint);
+        }
+        default:
+          break;
+      }
+    }
+    throw Exception(Type::GetInstance(value.GetTypeId())->ToString() +
+                    " is not coercable to BIGINT.");
+  }
+
+  static inline Value CastAsInteger(const Value &value) {
+    if (Type::GetInstance(TypeId::INTEGER)
+            ->IsCoercableFrom(value.GetTypeId())) {
+      if (value.IsNull())
+        return ValueFactory::GetIntegerValue((int32_t)PELOTON_INT32_NULL);
+      switch (value.GetTypeId()) {
+        case TypeId::TINYINT:
+          return ValueFactory::GetIntegerValue((int32_t)value.GetAs<int8_t>());
+        case TypeId::SMALLINT:
+          return ValueFactory::GetIntegerValue((int32_t)value.GetAs<int16_t>());
+        case TypeId::INTEGER:
+          return ValueFactory::GetIntegerValue(value.GetAs<int32_t>());
+        case TypeId::BIGINT: {
+          if (value.GetAs<int64_t>() > (int64_t)PELOTON_INT32_MAX ||
+              value.GetAs<int64_t>() < (int64_t)PELOTON_INT32_MIN)
+            throw Exception(ExceptionType::OUT_OF_RANGE,
+                            "Numeric value out of range.");
+          return ValueFactory::GetIntegerValue((int32_t)value.GetAs<int64_t>());
+        }
+        case TypeId::DECIMAL: {
+          if (value.GetAs<double>() > (double)PELOTON_INT32_MAX ||
+              value.GetAs<double>() < (double)PELOTON_INT32_MIN)
+            throw Exception(ExceptionType::OUT_OF_RANGE,
+                            "Numeric value out of range.");
+          return ValueFactory::GetIntegerValue((int32_t)value.GetAs<double>());
+        }
+        case TypeId::VARCHAR: {
+          std::string str = value.ToString();
+          int32_t integer = 0;
+          try {
+            integer = stoi(str);
+          } catch (std::out_of_range &e) {
+            throw Exception(ExceptionType::OUT_OF_RANGE,
+                            "Numeric value out of range.");
+          } catch(std::invalid_argument &e){
+            throw Exception("Invalid input syntax for integer: \'" + str + "\'");
```

Comment:
For some reason diff is improper for this commit. So I'm highlighting the lines modified.